pkg/debuginfo/debuginfod.go

```diff
@@ -0,0 +1,44 @@
+package debuginfo
+
+import (
+	"io"
+	"net/http"
+	"path"
+)
+
+var publicServer = "https://debuginfod.systemtap.org"
+
+type DebugInfodClient interface {
+	GetDebugInfo(buildid string) (io.ReadCloser, error)
+}
+
+type HttpDebuginfodClient struct {
+	UpstreamServer string //url
+}
+type ObjectStorageDebugInfodClientCache struct {
+	client DebugInfodClient
+}
+
+func NewHttpDebugInfoClient(serverUrl string) HttpDebuginfodClient {
+	return HttpDebuginfodClient{UpstreamServer: serverUrl}
+}
+
+func NewObjectStorageDebugInfodClientCache(h HttpDebuginfodClient) *ObjectStorageDebugInfodClientCache {
+	return &ObjectStorageDebugInfodClientCache{client: &h}
+}
+
+func (c *ObjectStorageDebugInfodClientCache) GetDebugInfo(buildid string) (io.ReadCloser, error) {
+	return c.client.GetDebugInfo(buildid)
```

Comment:
I think this still needs the part that actually caches the returned debug info files, right?